### PR TITLE
📖 : update ComponentConfig tutorial

### PR DESCRIPTION
**Pull request by Eileen-Yu**
### Description:
Update the scaffolded project of the Component Config tutorial based on the latest kubebuilder and the corresponding document: https://deploy-preview-3181--kubebuilder.netlify.app/component-config-tutorial/tutorial.html

### Motivation:
1. Keep the docs updated
2. This is an attempt to practice on https://github.com/kubernetes-sigs/kubebuilder/issues/2510

### CODE REVIEW

File: docs/book/src/component-config-tutorial/testdata/project/config/manager/controller_manager_config.yaml [hunk 1]
```diff
@@ -17,4 +17,3 @@ webhook:
 leaderElection:
   leaderElect: true
   resourceName: 80807133.tutorial.kubebuilder.io
-clusterName: example-test
```
Comment (camilamacedo86) [inline docs/book/src/component-config-tutorial/testdata/project/config/manager/controller_manager_config.yaml:20]: @Eileen-Yu we need to came back this line.

File: docs/book/src/component-config-tutorial/testdata/project/api/v2/projectconfig_types.go [hunk 2]
```diff
@@ -66,5 +65,5 @@ type ProjectConfigList struct {
 }
 
 func init() {
-	SchemeBuilder.Register(&ProjectConfig{}, &ProjectConfigList{})
+	SchemeBuilder.Register(&ProjectConfig{})
```
Comment (camilamacedo86) [inline docs/book/src/component-config-tutorial/testdata/project/api/v2/projectconfig_types.go:68]: We need to came back this line

File: docs/book/src/component-config-tutorial/testdata/project/api/v2/projectconfig_types.go [hunk 3]
```diff
@@ -52,8 +52,7 @@ type ProjectConfig struct {
 
 	// ControllerManagerConfigurationSpec returns the configurations for controllers
 	cfg.ControllerManagerConfigurationSpec `json:",inline"`
-
-	ClusterName string `json:"clusterName,omitempty"`
```
Comment (camilamacedo86) [inline docs/book/src/component-config-tutorial/testdata/project/api/v2/projectconfig_types.go:56]: We need to came back this line?

Also, after came back we need to run `make all` again.

**Review by camilamacedo86 — COMMENTED** [review]
Hi @Eileen-Yu,

That is great. Just a few nits. See that some customizations made on top were lost with the update. We need to diff and ensure that are addressed in the code again.

Also, after to do that we need just call `make` to run all targets and ensure that all is generated and updated. 

Then, we can get this one merged to update the docs with the latest changes. 
Thank you for looking on that.

**Comment by Eileen-Yu** [comment]
Hi @camilamacedo86 Thx for reviewing!

Just made those lines come back and called `make` to ensure the updates.

I'll open another PR to try to update the ComponentConfig tutorial programatically.

**Review by camilamacedo86 — APPROVED** [review]
Thank you for your contribution🥇 

/ok-to-test
/lgtm
/aproved

**Pull request merged**